slither/detectors/shadowing/return_local.py

```diff
@@ -0,0 +1,117 @@
+    # region wiki_exploit_scenario
+    WIKI_EXPLOIT_SCENARIO = """"
+```solidity
+pragma solidity ^0.8.0;
+
+contract Bug {
+    function shadowed() external view returns(uint value) {
+        uint value = 1;
+    } //returns 0
+
+    function unnamed() external view returns(uint) {
+        uint value = 1;
+    } //returns 0
+}
+```"""
+    # endregion wiki_exploit_scenario
+
+    WIKI_RECOMMENDATION = """
+    1. Name return variables inside "returns(...)".
+    2. Don't re-declare the type of return variables.
+    3. Add a `return` statement at the end of the function."""
+
+    def info(self, err_unnamed, err_shadowed): #4/4 (end) ↰
+        info = []
+        if len(err_unnamed)>0:
+            info.append(self.generate_result('💥💥💥 DETECTED UNNAMED RETURN VARIABLE 💥💥💥\n'))
+            info_tmp=''
+            for bug_location, unnamed_vars in err_unnamed.items():
+                for var in unnamed_vars:
+                    info_tmp+=f'\n      * "{var.type}"'
+                info.append(self.generate_result([
+                    '  * In: ', bug_location, '\n    * inside "returns(...)", set name for:', info_tmp, '\n'
+                ]))
+                info_tmp=''
+        if len(err_shadowed)>0:
+            info.append(self.generate_result('\n💥💥💥 DETECTED SHADOWED LOCAL VARIABLE 💥💥💥\n'))
+            info_tmp=''
+            for bug_location, shadow_var in err_shadowed.items():
+                for var in shadow_var:
+                    info_tmp+=f'\n    * from variable: "{var}"'
+                    info_tmp+=f'\n     * remove type declaration: "{var.type}"'
+                info.append(self.generate_result(['  * Inside: ', bug_location, info_tmp, '\n']))
+                info_tmp = ''
+        return info
+
+    def detect_shadowed_local_vars(self, function): #3/4 ↑
+        shadowed_local_vars = []
+
+        for return_var in function.returns:
+            for local_var in function.variables:
+                try: # to prevent e.g. wrong format errors
+                    #get shadowing info, e.g., from name_scope_0; extract _scope_0
+                    scope_part = re.sub(f'^{return_var.name}', "", local_var.name)
+                    #save "return_var" if shadows "local_var"
+                    if re.search('^_scope_[0-9]$', scope_part):
+                        shadowed_local_vars.append(return_var) #(return=local without scope)
+                except Exception as e:
+                    print(e); continue
+
+        return shadowed_local_vars
+
+    def detect_unnamed_return_vars(self, function): #2/4 ↑
+        unnamed_return_vars = []
+
+        for var in function.returns:
+            if str(var)=='':
+                unnamed_return_vars.append(var)
+
+        return unnamed_return_vars
+
+    def no_return_statment_in(self, function): #1/4 ↑
+        if len(function.nodes)==0: return False #ignore inherited interfaces
+        for node in function.nodes:
+            if function.nodes and str(node.type)=='RETURN':
```

Comment:
`function.nodes` is not needed here. Additionally, we should check against `NodeType.RETURN` instead of checking the `str` to make sure its robust